scripts/albs-1147.py

```diff
@@ -0,0 +1,181 @@
+
+            start_context = max(0, i1 - 3)
+            end_context = min(len(original_words), i2 + 3)
+            original_context = ' '.join(original_words[start_context:i1]) \
+                               + ' 【' + original_change + '】 ' \
+                               + ' '.join(original_words[i2:end_context])
+
+            start_context = max(0, j1 - 3)
+            end_context = min(len(new_words), j2 + 3)
+            new_context = ' '.join(new_words[start_context:j1]) \
+                          + ' 【' + new_change + '】 ' \
+                          + ' '.join(new_words[j2:end_context])
+
+            log.info(f'... {original_context} ➔ {new_context}')
+
+
+async def main(write=False):
+    ignore_parts = [
+        'access.redhat.com',
+        'container-tools:rhel8',
+        'eclipse:rhel8',
+        'go-toolset:rhel8',
+        'jmc:rhel8',
+        'llvm-toolset:rhel8',
+        'rust-toolset:rhel8',
+        'virt:rhel',
+        'virt-devel:rhel',
+        'ansible-collection-redhat-rhel_mgmt',
+        'ansible-collection-AlmaLinux-rhel_mgmt',
+        'rhel-system-roles',
+        'lorax-templates-rhel',
+        'redhat-rpm-config',
+        'kmod-redhat-oracleasm',
+        'rhel8 stream'
+    ]
+    search_parts = [
+        'rhel',
+        'red hat',
+        'redhat'
+    ]
+    affected_records = {}
+
+    with get_pulp_db() as session:
+        result = session.execute(
+            select(UpdateRecord).where(
+                or_(
+                    *[UpdateRecord.description.ilike(f'%{part}%') for part in search_parts],
+                    *[UpdateRecord.title.ilike(f'%{part}%') for part in search_parts],
+                ),
+                not_(
+                    or_(
+                        *[UpdateRecord.description.ilike(f'%{part}%') for part in ignore_parts],
+                        *[UpdateRecord.title.ilike(f'%{part}%') for part in ignore_parts],
+                    )
+                )
+            )
+        )
+        records = result.scalars().all()
+
+        log.info(f'Found {len(records)} records in Pulp\'s \'rpm_updaterecord\' table.')
+        record: UpdateRecord
+        for record in records:
+            log.info(f'{record.id} - {record.title}')
+            debranded_title = debrand_description_and_title(record.title)
+            debranded_description = debrand_description_and_title(record.description)
+            log_differences(record.title, debranded_title)
+            log_differences(record.description, debranded_description)
+            record.title = debranded_title
+            record.description = debranded_description
+            affected_records[record.id] = {
+                'title': record.title,
+                'description': record.description
+            }
+            record.updated_date = datetime.datetime.utcnow().strftime(
+                "%Y-%m-%d %H:%M:%S",
+            )
+
+        if write:
+            session.commit()
```

Comment:
I'm confused about using `session.commit` inside a context manager. Doesn't it actually save and commit changes to db when exiting the context?

Reply:
No, it doesn't — changes are fixed only after the `commit()` execution.